File: lib/constant.ts

import {
    PublicKey,
} from "@solana/web3.js";

export const GLOBAL_AUTHORITY_SEED = "global-sol-token-authority";
export const VAULT_AUTHORITY_SEED = "vault-space-authority";
export const USER_POOL_SEED = "user-pool-facon-seed";       

export const PROGRAM_ID = new PublicKey("EE6PYbaGBMvvUs5GFppdCTJ3mGWYzWAzFxdeMs268VLy");
    
export const MILSECS_IN_DAY = (1000 * 60 * 60 * 24);

export const ELMNT_ADDRESS = new PublicKey("4EKbguCsFW3TfpWs8xDnUKFLZheB3rnoRViApjxCicoq");
export const ELMNT_DECIMAL = 1000000;

// Define the minimum lock duration in seconds (1 month = 30 days)
export const MIN_LOCK_DURATION_SECONDS = 30 * 24 * 60 * 60;

File: programs/sol_token_staking/src/constant.rs
use solana_program::native_token::LAMPORTS_PER_SOL;

pub const GLOBAL_AUTHORITY_SEED: &str = "global-sol-token-authority";
pub const USER_AUTHORITY_SEED: &str = "user-v1-authority";
pub const VAULT_AUTHORITY_SEED: &str = "vault-space-authority";
pub const USER_POOL_SEED: &str = "user-pool-facon-seed";

pub const DAY_SECONDS: u64 =  60 * 60 * 24;

// pub const ELMNT_ADDRESS: &str = "5CY4inXAWEKDENqJ5ZLNaTYX8gzjHZNXimuj7VmFmVi6";
pub const ELMNT_ADDRESS: &str = "5CY4inXAWEKDENqJ5ZLNaTYX8gzjHZNXimuj7VmFmVi6";

pub const ELMNT_DECIMAL: u64 = 1000000;

pub const SOL_AMOUNT: u64 = LAMPORTS_PER_SOL / 10;
pub const TOKEN_AMOUNT: u64 = ELMNT_DECIMAL * 1;
pub const TOKEN_LOW_RATE: u64 = ELMNT_DECIMAL / 10;
pub const SOL_LOW_RATE: u64 = ELMNT_DECIMAL / 10;

pub const LOCK_PERIOD: [i64; 3] = [120, 240, 300];
pub const REWARD_RATE: [u64; 3] = [0, 10, 20];